sycl/source/detail/scheduler/commands.hpp

```diff
@@ -570,7 +570,8 @@ pi_int32 enqueueImpKernel(
     const std::shared_ptr<detail::kernel_impl> &MSyclKernel,
     const std::string &KernelName, const detail::OSModuleHandle &OSModuleHandle,
     std::vector<RT::PiEvent> &RawEvents, RT::PiEvent *OutEvent,
-    const std::function<void *(Requirement *Req)> &getMemAllocationFunc);
+    const std::function<std::optional<void *>(Requirement *Req)>
```

Comment:
`std::optional` seems redundant for a pointer type, we should probably just use `nullptr` to denote a "missing" value instead. The same goes for `getAllocaReq`.